hudi-spark-datasource/hudi-spark2/src/main/scala/org/apache/spark/sql/adapter/Spark2Adapter.scala

```diff
@@ -86,4 +87,14 @@ class Spark2Adapter extends SparkAdapter {
   override def parseMultipartIdentifier(parser: ParserInterface, sqlText: String): Seq[String] = {
     throw new IllegalStateException(s"Should not call ParserInterface#parseMultipartIdentifier for spark2")
   }
+
+  /**
+   * Combine [[PartitionedFile]] to [[FilePartition]] according to `maxSplitBytes`.
+   */
+  override def getFilePartitions(
+      sparkSession: SparkSession,
+      partitionedFiles: Seq[PartitionedFile],
+      maxSplitBytes: Long): Seq[FilePartition] = {
+    HoodieDataSourceHelper.getFilePartitions(sparkSession, partitionedFiles, maxSplitBytes)
```

Comment:
It'd actually be the other way around: `Spark2Adapter` should hold the impl copied from Spark3, while `HoodieDataSourceHelper` will be invoking it

Reply:
understood.